src/aspire/source/simulation.py

```diff
@@ -48,12 +48,11 @@ def __init__(self, L=8, n=1024, vols=None, states=None, filters=None,
         self.offsets = offsets
         self.amplitudes = amplitudes
         self.angles = angles
-        self.C = C
         if vols is None:
-            self.vols = self._gaussian_blob_vols(L=self.L, C=self.C, seed=seed)
+            self.vols = self._gaussian_blob_vols(L=self.L, C=C, seed=seed)
         else:
             self.vols = vols
```

Comment:
Do we need to assert C equals the number of volumes for other code to be okay? (like line 40?).

Reply:
No I think we're fine. If this block is entered, `vols` is None, and there's nothing to check. `self.C` gets initialized outside this if-else block once `self.vols` is initialized.

Reply:
If I pass vols of length k, and C=2 (by default for example) then line 40 above would know only the local C=2, not self.C = k which is assigned just below. Without digging in deeper I'm not sure if that is a problem.

Reply:
(perhaps I should have put the comment on l40, but it wasn't in this changeset).

Reply:
I think this block and assignment just need to move up above l40, that's all.